src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIImageClient.cs

```diff
@@ -0,0 +1,250 @@
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI;
+using OpenAI.Images;
+
+#pragma warning disable S3011 // Reflection should not be used to increase accessibility of classes, methods, or fields
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="IImageClient"/> for an OpenAI <see cref="OpenAIClient"/> or <see cref="ImageClient"/>.</summary>
+internal sealed class OpenAIImageClient : IImageClient
+{
+    /// <summary>Metadata about the client.</summary>
+    private readonly ImageClientMetadata _metadata;
+
+    /// <summary>The underlying <see cref="ImageClient" />.</summary>
+    private readonly ImageClient _imageClient;
+
+    /// <summary>Initializes a new instance of the <see cref="OpenAIImageClient"/> class for the specified <see cref="ImageClient"/>.</summary>
+    /// <param name="imageClient">The underlying client.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="imageClient"/> is <see langword="null"/>.</exception>
+    public OpenAIImageClient(ImageClient imageClient)
+    {
+        _ = Throw.IfNull(imageClient);
+
+        _imageClient = imageClient;
+
+        // https://github.com/openai/openai-dotnet/issues/215
+        // The endpoint and model aren't currently exposed, so use reflection to get at them, temporarily. Once packages
+        // implement the abstractions directly rather than providing adapters on top of the public APIs,
+        // the package can provide such implementations separate from what's exposed in the public API.
+        Uri providerUrl = typeof(ImageClient).GetField("_endpoint", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(imageClient) as Uri ?? OpenAIClientExtensions.DefaultOpenAIEndpoint;
+
+        _metadata = new("openai", providerUrl, _imageClient.Model);
+    }
+
+    /// <inheritdoc />
+    public async Task<ImageResponse> GenerateImagesAsync(ImageRequest request, ImageOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(request);
+
+        string? prompt = request.Prompt;
+        _ = Throw.IfNull(prompt);
+
+        // If the request has original images, treat this as an edit operation
+        if (request.OriginalImages != null)
+        {
+            ImageEditOptions editOptions = ToOpenAIImageEditOptions(options);
+            string? fileName = null;
+            Stream? imageStream = null;
+
+            foreach (AIContent originalImage in request.OriginalImages)
+            {
+                if (imageStream is not null)
+                {
+                    // We only support a single image for editing.
+                    Throw.ArgumentException(
+                        "Only a single image can be provided for editing.",
+                        nameof(request));
```

Comment:
I think it'd be better to just ignore subsequent ones. It's no different than if the model just disregarded some of the input. And throwing in a case like this makes it harder for the interface to be used in a provider-agnostic way.